elemental/client.py

```diff
@@ -87,9 +92,14 @@ def send_request(self, http_method: str, url: str, headers: Dict[str, str],
         except requests.exceptions.RequestException as e:
             raise InvalidRequest(f"{http_method}: {url} failed\n{e}")
         if response.status_code not in (200, 201):
-            raise InvalidResponse(
-                f"{http_method}: {url} failed\nResponse: "
-                f"{response.status_code}\n{response.text}")
+            if response.status_code == 404:
+                raise NotFound(
+                    f"{http_method}: {url} failed\nResponse: "
+                    f"{response.status_code}\n{response.text}")
+            else:
+                raise InvalidResponse(
+                    f"{http_method}: {url} failed\nResponse: "
+                    f"{response.status_code}\n{response.text}")
```

Comment:
Not really an issue, but I'd rather avoid the nested conditions. We could perform those at the same level
```python
if response.status_code == 404:
   raise NotFound(...)
if response.status_code not in (200, 201):
   raise InvalidResponse(...)
return response
```